src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/utils/OwnersAndOfficersDataBlockValidator.java

```diff
@@ -0,0 +1,152 @@
+            if (hasGovernmentOrPublicAuthorityBeneficialOwnersPresent(beneficialOwnerGovernmentOrPublicAuthorityDtoList)) {
+                beneficialOwnerGovernmentOrPublicAuthorityValidator.validate(beneficialOwnerGovernmentOrPublicAuthorityDtoList, errors, loggingContext);
+            }
+            List<ManagingOfficerIndividualDto> managingOfficersIndividualDtoList = overseasEntitySubmissionDto.getManagingOfficersIndividual();
+            if (hasIndividualManagingOfficersPresent(managingOfficersIndividualDtoList)) {
+                managingOfficerIndividualValidator.validate(managingOfficersIndividualDtoList, errors, loggingContext);
+            }
+            List<ManagingOfficerCorporateDto> managingOfficersCorporateDtoList = overseasEntitySubmissionDto.getManagingOfficersCorporate();
+            if (hasCorporateManagingOfficersPresent(managingOfficersCorporateDtoList)) {
+                managingOfficerCorporateValidator.validate(managingOfficersCorporateDtoList, errors, loggingContext);
+            }
+        }
+    }
+
+    private boolean isCorrectCombinationOfOwnersAndOfficersForStatement(OverseasEntitySubmissionDto overseasEntitySubmissionDto, Errors errors, String loggingContext) {
+        switch(overseasEntitySubmissionDto.getBeneficialOwnersStatement()) {
+            case ALL_IDENTIFIED_ALL_DETAILS:
+                 if (!hasBeneficialOwners(overseasEntitySubmissionDto)) {
+                     logValidationErrorMessage(errors, loggingContext, String.format("%s for statement that all can be identified", MISSING_BENEFICIAL_OWNER));
+                     return false;
+                 }
+                 if (hasManagingOfficers(overseasEntitySubmissionDto)) {
+                     logValidationErrorMessage(errors, loggingContext, String.format("%s for statement that all can be identified", INCORRECTLY_ADDED_MANAGING_OFFICER));
+                     return false;
+                 }
+                 break;
+            case SOME_IDENTIFIED_ALL_DETAILS:
+                if (!hasBeneficialOwners(overseasEntitySubmissionDto)) {
+                    logValidationErrorMessage(errors, loggingContext, String.format("%s for statement that some can be identified", MISSING_BENEFICIAL_OWNER));
+                    return false;
+                }
+                break;
+            case NONE_IDENTIFIED:
+                if (!hasManagingOfficers(overseasEntitySubmissionDto)) {
+                    logValidationErrorMessage(errors, loggingContext, String.format("%s for statement that none can be identified", MISSING_MANAGING_OFFICER));
+                    return false;
+                }
+                if (hasBeneficialOwners(overseasEntitySubmissionDto)) {
+                    logValidationErrorMessage(errors, loggingContext, String.format("%s for statement that none can be identified", INCORRECTLY_ADDED_BENEFICIAL_OWNER));
+                    return false;
+                }
+                break;
+        }
```

Comment:
if the user supplies a non standard value for the statement eg "NONE", would it pass the validation? I think it would be good to have a 'default' in the switch so that if the value is not matched against our expected statement values then we should return validation message - 'Invalid statement supplied'?

Possibly add a similar check to the beneficialOwnersStatementValidator.validate() to block any random statement values (if it doesn't already)

Reply:
As this is an enum I'm not sure this is going to help us.

Reply:
just tried it and api will fail to parse it if it's not one of the enum values so should be ok. I think it's good practice to have the default so you can leave that in there.